git_machete/client.py

```diff
@@ -2035,7 +2035,14 @@ def retarget_github_pr(self, head: LocalBranchShortName) -> None:
         else:
             print(fmt(f'The base branch of PR #{pr.number} is already `{new_base}`'))
 
-    def __derive_remote_and_github_org_and_repo(self) -> RemoteAndOrganizationAndRepository:
+    def _filter_remote_based_on_the_branch_tacking_data(self, remotes: List[str], branch: LocalBranchShortName) -> List[str]:
+        for remote in remotes:
+            self.__git.fetch_remote(remote)
+        remote_branches = self.__git.get_remote_branches()
+        remotes_containing_branch: List[str] = [remote for remote in remotes if f'{remote}/{branch}' in remote_branches]
```

Comment:
Hmmm this logic for finding a single remote containing the given branch... this feels like it's already implemented somewhere and shouldn't be reimplemented here 🤔

Reply:
See `git_machete.git_operations.GitContext.get_combined_remote_for_fetching_of_branch` and the surrounding/related methods

Reply:
Good spot! `get_inferred_remote_for_fetching_of_branch` shouldn't work this way 🤔 lemme open an issue to rectify it (in fact, make it more aligned with what you've implemented here, with `len(remotes_containing_branch) == 1`: only consider the remote branch `R/X` to be an adequate _inferred counterpart_ for `X` if only **one** remote `R` contains a branch named `X`)

Reply:
I will create similar issue as this one for `discover`

Reply:
+1, thanks for sorting that out, that was my exact action plan... just one remark, already stated elsewhere:

> `get_strict_remote_for_fetching_of_branch` ... that returns remote but if there are more than one it will return remote that was added last

Nopes, `get_strict_remote_for_fetching_of_branch` is doing what it should AFAICS! but `get_inferred_...` methods were erroneous indeed